osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -78,49 +79,50 @@ std::string accessPermsToStr(const unsigned long pmask) {
 }
 
 // helper function to get account/group name from trustee
-std::string trusteeToStr(const TRUSTEE& trustee) {
+std::string trusteeToStr(const TRUSTEEW& trustee) {
   // Max username length for Windows
   const unsigned long maxBuffSize = 256;
   unsigned long sizeOut = maxBuffSize;
-  char name[maxBuffSize];
-  char domain[maxBuffSize];
+  WCHAR name[maxBuffSize];
+  WCHAR domain[maxBuffSize];
   SID_NAME_USE accountType;
 
   switch (trustee.TrusteeForm) {
   case TRUSTEE_IS_SID: {
     // get the name from the SID
     PSID psid = trustee.ptstrName;
-    auto r = LookupAccountSid(
+    auto r = LookupAccountSidW(
         nullptr, psid, name, &sizeOut, domain, &sizeOut, &accountType);
     if (r == FALSE) {
       VLOG(1) << "LookupAccountSid error: " << GetLastError();
       return "";
     } else {
-      return name;
+      return wstringToString(name);
     }
   }
   case TRUSTEE_IS_NAME:
     // get the name from ptstrName
-    return trustee.ptstrName;
+    return wstringToString(trustee.ptstrName);
   case TRUSTEE_BAD_FORM:
     // Indicates a trustee form that is not valid.
     // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379638(v=vs.85).aspx
     return "Invalid";
   case TRUSTEE_IS_OBJECTS_AND_SID: {
     // ptstrName member is a pointer to an OBJECTS_AND_SID struct
     auto psid = reinterpret_cast<POBJECTS_AND_SID>(trustee.ptstrName)->pSid;
-    auto r = LookupAccountSid(
+    auto r = LookupAccountSidW(
         nullptr, psid, name, &sizeOut, domain, &sizeOut, &accountType);
     if (r == FALSE) {
       VLOG(1) << "LookupAccountSid error: " << GetLastError();
       return "";
     } else {
-      return name;
+      return wstringToString(name);
     }
   }
   case TRUSTEE_IS_OBJECTS_AND_NAME:
     // ptstrName member is a pointer to an OBJECTS_AND_NAME struct
-    return reinterpret_cast<OBJECTS_AND_NAME_*>(trustee.ptstrName)->ptstrName;
+    return wstringToString(
+        reinterpret_cast<OBJECTS_AND_NAME_W*>(trustee.ptstrName)->ptstrName);
```

Comment:
Mmm, `pObjectsAndName` is only available if over RPC (i.e., `__midl` case). Here's the spec:

```
typedef struct _TRUSTEE_W
{
    struct _TRUSTEE_W          *pMultipleTrustee;
    MULTIPLE_TRUSTEE_OPERATION  MultipleTrusteeOperation;
    TRUSTEE_FORM                TrusteeForm;
    TRUSTEE_TYPE                TrusteeType;
#ifdef __midl
    [switch_is(TrusteeForm)]
    union
    {
    [case(TRUSTEE_IS_NAME)]
        LPWSTR                  ptstrName;
    [case(TRUSTEE_IS_SID)]
        SID                    *pSid;
    [case(TRUSTEE_IS_OBJECTS_AND_SID)]
        OBJECTS_AND_SID        *pObjectsAndSid;
    [case(TRUSTEE_IS_OBJECTS_AND_NAME)]
        OBJECTS_AND_NAME_W     *pObjectsAndName;
    };
#else
    // This member is not null-terminated as it may be used to hold strings, which are null-terminated or 
    // SID, which are not null-terminated.
    LPWCH                       ptstrName;
#endif
} TRUSTEE_W, *PTRUSTEE_W, TRUSTEEW, *PTRUSTEEW;
```

`__midl` isn't defined in this case.

Reply:
I see, I've checked the docs but indeed the headers are more involved.
What set me off is the fact that reinterpret_cast in this case is a strict aliasing violation and so it's UB.
That been said, I see it was already existing and compilers relax the strict aliasing rule sometimes, so I won't ask you to fix it now.
I'll open a separate issue.

Reply:
Ok-- I will resolve this conversation. :)